### ISSUE TITLE
App crash on 0.30.0 when branch.registerFacebookDeepLinkingClass

### ISSUE THREAD

**Issue by shial4** [BranchMetrics/ios-branch-deep-linking-attribution#971]
App crashes on branch.registerFacebookDeepLinkingClass the reason is asynchronous operation which is no longer main thread. 

```
if let branch = Branch.getInstance() {
            branch.registerFacebookDeepLinkingClass(FBSDKAppLinkUtility.self)
            branch.suppressWarningLogs()
            branch.delayInitToCheckForSearchAds()
            branch.initSession(launchOptions: launchOptions, andRegisterDeepLinkHandler: { params, error in
                if error == nil {
                    // params are the deep linked params associated with the link that the user clicked -> was re-directed to this app
                    // params will be empty if no data found
                    // ... insert custom logic here ...
                    
                    if let params = params {
                        if  let iosUrl = self.branchUrlParams.filter({item in params[item] != nil}).map({item in params[item]}).first as? String, let deepLinkUrl = URL(string: iosUrl){
                            if DeeplinkHandler.shared.canHandleDeeplink(withUrl: deepLinkUrl) {
                                DeeplinkHandler.shared.handleDeeplink(withUrl: deepLinkUrl)
                                return
                            }
                        }
                        DeepLinker.shared.handleDeepLinking(params)
                    }
                    
                    if let params = params as? [String: AnyObject] {
                        print("params: %@", params)
                    }
                }
            })
        }
```

**Comment by alfredcc** [comment]
0.30.0 crash +1

**Comment by roth-mike** [comment]
0.30.0 crash +1

**Comment by alex-kovalenko** [comment]
I'm having the same issue.

**Comment by echo-branch** [comment]
Fixed in 0.30.1

I failed to remove my mock and test with the real FB object.  :(
